compiler/optimizations/removeUnnecessaryAutoCopyCalls.cpp

```diff
@@ -85,9 +85,12 @@ void RemoveUnnecessaryAutoCopyCalls::process(FnSymbol* fn)
     Type* lhsType = NULL;
     Type* rhsType = actual->typeInfo();
 
-    CallExpr* move = toCallExpr(call->parentExpr);
-    INT_ASSERT(isMoveOrAssign(move));
-    lhsType = move->get(1)->typeInfo();
+    CallExpr* callParent = toCallExpr(call->parentExpr);
+    if(callParent->isPrimitive(PRIM_TASK_INDEPENDENT_SVAR_CAPTURE)) {
```

Comment:
Could you add a comment here explaining why this was necessary?

Reply:
The subsequent code in this function assumes that `callParent` is a move or assign expr and processes accordingly. I thought this pass wasn't relevant for a use of `chpl__initCopy` in `PRIM_TASK_INDEPENDENT_SVAR_CAPTURE` so I just skipped processing altogether, which seemed to resolve my paratest failures.

Although now thinking about it some more I'm not actually sure if that's true or not. I tried changing the code to instead set `callParent` to the parentexpr of `PRIM_TASK_INDEPENDENT_SVAR_CAPTURE` and then proceed with processing. This seems to work (paratests) pass.

I've added a comment. So now it reads like this:

```
// When processing 'in' intent variables passed to foreach loops, we
// created a task independent captured version of the variable. The
// exact way to do that depends on whether the loop is ultimately
// cpu or gpu bound so at this point of compilation we represent
// this using an assignment to PRIM_TASK_INDEPENDENT_SVAR_CAPTURE.
// In cases where the 'in' intent'ed variable is an object we
// may have AST like this:
//
//  task_ind_x = PRIM_TASK_INDEPENDENT_SVAR_CAPTURE(chpl__initCopy(x))
//
// To proceed, we basically ignore the primitive on the RHS of the
// assignment and process as if it were written
// `task_ind_x = chpl_initCopy(x)`.
if (callParent->isPrimitive(PRIM_TASK_INDEPENDENT_SVAR_CAPTURE)) {
  callParent = toCallExpr(callParent->parentExpr);
}
```